Automattic/wp-calypso
# Gutenboarding: Signup modal appears above existing page content, behaves more like a modal

#### Changes proposed in this Pull Request

* Move signup modal code out of `onboarding-block/` and into `components/`, to drive home this is a generic modal that can appear at many places in the flow
* Opening modal no longer changes route, it should appear over whatever content the user is currently looking at
* TypeScript shenanigans to get the correct type when dispatching actions that use generator functions
* Remove the JSON dump that appeared in the modal when signup was successful

#### Testing instructions



* In incognito tab go to `/gutenboarding`
* Go through flow and click "Create my site"
* Modal opens over the page layout selector
* Esc/clicking outside/clicking cross, should close signup modal
* Clicking "Create my site" again should reopen dialog
* Esc/clicking outside, in the layout selector should still close the layout selector
